cpu/efm32/periph/i2c.c

```diff
@@ -54,28 +56,21 @@ static mutex_t i2c_lock[I2C_NUMOF];
  */
 static int _transfer(i2c_t dev, I2C_TransferSeq_TypeDef *transfer)
 {
-    bool busy = true;
-    uint32_t timeout = I2C_TIMEOUT;
+    uint32_t counter = EFM32_I2C_TIMEOUT;
```

Comment:
Shouldn't the timeout depend on the length of the transfer?

Reply:
Ideally, yes. And also work with clock stretching.

But I have taken that value from the kit driver and it has been in this driver since the beginning.

Reply:
Wouldn't we only have to wait for the first successful state transition, then we can block on a mutex until the transfer is done - that would make for a very small timeout that is also independent of transfer length.